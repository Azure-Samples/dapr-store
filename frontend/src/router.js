import VueRouter from 'vue-router'

import ProductCatalog from './components/ProductCatalog'
import ProductOffers from './components/ProductOffers'
import Home from './components/Home'

const router = new VueRouter({
  routes: [
    {
      path: '/catalog',
      component: ProductCatalog
    },
    {
      path: '/offers',
      component: ProductOffers
    },
    {
      path: '/',
      component: Home
    }
  ]
})

export default router