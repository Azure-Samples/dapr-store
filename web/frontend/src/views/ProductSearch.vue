<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - Show product results from a search
// ----------------------------------------------------------------------------
-->

<template>
  <div>
    <h1>Search results for: <i>{{ this.$route.params.query }}</i></h1>
    <error-box :error="error" />
    <product-list v-if="!error" :products="products" />
  </div>
</template>

<script>
import api from '../services/api'
import ProductList from '../components/ProductList'
import ErrorBox from '../components/ErrorBox'

export default {
  name: 'ProductSearch',

  components: {
    'product-list': ProductList,
    'error-box': ErrorBox
  },

  data() {
    return {
      products: null,
      error: null
    }
  },

  watch: {
    async $route() {
      this.doSearch()
    }
  },

  async mounted() {
    this.doSearch()
  },

  methods: {
    async doSearch() {
      try {
        this.products = null
        this.products = await api.productSearch(this.$route.params.query)
      } catch (err) {
        this.error = err
      }
    }
  }
}
</script>
