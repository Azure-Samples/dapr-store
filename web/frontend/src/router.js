// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - client side routes
// ----------------------------------------------------------------------------

import { createRouter, createWebHistory } from 'vue-router'

import auth from './services/auth'

import ProductCatalog from './views/ProductCatalog.vue'
import ProductOffers from './views/ProductOffers.vue'
import ProductSearch from './views/ProductSearch.vue'
import ProductSingle from './views/ProductSingle.vue'
import ViewOrder from './views/ViewOrder.vue'
import Login from './views/Login.vue'
import Account from './views/Account.vue'
import Home from './views/Home.vue'
import About from './views/About.vue'
import Cart from './views/Cart.vue'

const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      name: 'home',
      path: '/',
      component: Home
    },
    {
      name: 'login',
      path: '/login',
      component: Login
    },
    {
      name: 'account',
      path: '/account',
      component: Account,
      beforeEnter: signedInCheck
    },
    {
      name: 'cart',
      path: '/cart',
      component: Cart,
      beforeEnter: signedInCheck
    },
    {
      name: 'about',
      path: '/about',
      component: About
    },
    {
      name: 'search',
      path: '/search/:query',
      component: ProductSearch
    },
    {
      name: 'catalog',
      path: '/catalog',
      component: ProductCatalog
    },
    {
      name: 'offers',
      path: '/offers',
      component: ProductOffers
    },
    {
      name: 'single-product',
      path: '/product/:id',
      component: ProductSingle
    },
    {
      name: 'view-order',
      path: '/order/:id',
      component: ViewOrder
      //beforeEnter: signedInCheck
    }
  ]
})

function signedInCheck(to, from, next) {
  const user = auth.user()
  if (!user || !user.username) {
    next('/login')
  } else {
    next()
  }
}

export default router
