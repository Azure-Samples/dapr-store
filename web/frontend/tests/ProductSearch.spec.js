import { describe, it, vi, expect } from 'vitest'
import { mount, flushPromises } from '@vue/test-utils'
import router from '@/router'

import ProductSearch from '@/views/ProductSearch.vue'

vi.mock('@/services/api')

describe('ProductSearch.vue', () => {
  it('renders search for Ascot', async () => {
    router.push('/search/Ascot')
    await router.isReady()

    const wrapper = mount(ProductSearch, {
      global: {
        plugins: [router]
      }
    })

    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
