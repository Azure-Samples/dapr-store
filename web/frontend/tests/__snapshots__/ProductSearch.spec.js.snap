// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductSearch.vue > renders search for Ascot 1`] = `
"<div>
  <h1> Search results for: <i>Ascot</i></h1>
  <!--v-if-->
  <div data-v-32d040e6="">
    <!--v-if-->
    <!--v-if-->
    <div data-v-32d040e6="" class="card">
      <div data-v-32d040e6="" class="card-body">
        <div data-v-32d040e6="" class="row">
          <div data-v-32d040e6="" class="col"><a data-v-32d040e6="" href="/product/prd3" class="">
              <div data-v-32d040e6="" class="card-title">Paisley Cravat Ascot Tie</div>
            </a>
            <div data-v-32d040e6="" class="card-text">Black &amp; Red Paisley Cravat <h4 data-v-32d040e6="" class="mt-4">Â£11.2</h4>
            </div><button data-v-32d040e6="" disabled="" href="#" class="btn btn-primary d-none d-md-inline"><i data-v-32d040e6="" class="fa-solid fa-basket-shopping"></i> &nbsp; Add to Cart </button>
          </div>
          <div data-v-32d040e6="" class="col flex-grow-0 d-none d-md-block">
            <div data-v-32d040e6="" class="product-img">
              <!--v-if--><a data-v-32d040e6="" href="/product/prd3" class=""><img data-v-32d040e6="" src="/img/catalog/3.jpg"></a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div data-v-32d040e6="" class="position-fixed top-0 start-50 translate-middle-x p-3" style="z-index: 11;">
      <div data-v-32d040e6="" class="toast hide" role="alert">
        <div data-v-32d040e6="" class="d-flex">
          <div data-v-32d040e6="" class="toast-body fs-4"> was added to your cart!</div><button data-v-32d040e6="" type="button" class="btn-close me-2 m-auto" data-bs-dismiss="toast" aria-label="Close"></button>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
